### PR TITLE
adding the aria-required attributes to textbox component

### PR DESCRIPTION
**Pull request by amazingphilippe**
# Summary | Résumé

Fixes https://github.com/cds-snc/notification-planning/issues/66

Fixes https://github.com/cds-snc/notification-planning/issues/67

But also makes sure all textboxes have the aria-required attribute set to true as well. 

# Test instructions | Instructions pour tester la modification

1. Inspect any form element. It should have both the `required` and `aria-required="true"` attributes. Both are important

# Help requested | Aide requise

I want to crowd source this. **Do we have any optional fields?**

# Todo

- [x] Go in and mark optional fields as optional. I found a few while casually reading forms.py

# Reviewer checklist | Liste de vérification du réviseur

This is a suggested checklist of questions reviewers might ask during their
review | Voici une suggestion de liste de vérification comprenant des questions
que les réviseurs pourraient poser pendant leur examen :

- [ ] Does this meet a user need? | Est-ce que ça répond à un besoin utilisateur?
- [ ] Is it accessible? | Est-ce que c’est accessible?
- [ ] Is it translated between both offical languages? | Est-ce dans les deux
      langues officielles?
- [ ] Is the code maintainable? | Est-ce que le code peut être maintenu?
- [ ] Have you tested it? | L’avez-vous testé?
- [ ] Are there automated tests? | Y a-t-il des tests automatisés?
- [ ] Does this cause automated test coverage to drop? | Est-ce que ça entraîne
      une baisse de la quantité de code couvert par les tests automatisés?
- [ ] Does this break existing functionality? | Est-ce que ça brise une
      fonctionnalité existante?
- [ ] Should this be split into smaller PRs to decrease change risk? | Est-ce
      que ça devrait être divisé en de plus petites demandes de tirage (« pull
      requests ») afin de réduire le risque lié aux modifications?
- [ ] Does this change the privacy policy? | Est-ce que ça entraîne une
      modification de la politique de confidentialité?
- [ ] Does this introduce any security concerns? | Est-ce que ça introduit des
      préoccupations liées à la sécurité?
- [ ] Does this significantly alter performance? | Est-ce que ça modifie de
      façon importante la performance?
- [ ] What is the risk level of using added dependencies? | Quel est le degré de
      risque d’utiliser des dépendances ajoutées?
- [ ] Should any documentation be updated as a result of this? (i.e. README
      setup, etc.) | Faudra-t-il mettre à jour la documentation à la suite de ce
      changement (fichier README, etc.)?

### CODE REVIEW

**Comment by amazingphilippe** [comment]
Looking through forms.py, I have not seen any fields that should be optional. Therefore all should have the aria-required="true" attribute.

File: app/templates/components/textbox.html [hunk 1]
```diff
@@ -55,6 +55,7 @@
       class=field_class,
       data_module='highlight-tags' if highlight_tags else '',
       rows=rows|string,
+      aria_required='false' if optional else 'true',
```
Comment (jzbahrai) [inline app/templates/components/textbox.html:58]: given your latest comment should we change this?

Comment (amazingphilippe) [reply app/templates/components/textbox.html:58]: Does it make more sense to only add this line if we need optional fields in the future?

Comment (amazingphilippe) [reply app/templates/components/textbox.html:58]: But before we hardcode this, I'd like to know how all our fields end up with the "required" attribute. (different from aria-required="true")

Comment (jimleroyer) [reply app/templates/components/textbox.html:58]: Approving the PR but hope we resolved this thread.

**Review by jimleroyer — APPROVED** [review]

File: app/templates/components/textbox.html (hunk 1 above)
Comment (amazingphilippe) [reply app/templates/components/textbox.html]: I removed the optional conditional.
1. We don't have optional fields using the textbox component
2. I forgot to add the "optional" parameter to the macro signature. 

It'll make this PR be simpler and we'll do a better implementation the day we'll decide to use textbox for optional fields.

**Pull request merged**